VideoApp/VideoApp/Video/VideoAppCameraSource.swift

```diff
@@ -0,0 +1,159 @@
+        
+        if let cameraSource = cameraSource {
+            localVideoTrack = LocalVideoTrack(source: cameraSource, enabled: true, name: "camera")
+            startCameraSource()
+        }
+    }
+    
+    private func startCameraSource() {
+        guard let cameraSource = cameraSource else { return }
+
+        let targetSize: CMVideoDimensions
+        let cropRatio: CGFloat
+        let frameRate: UInt
+        
+        switch appSettingsStore.videoCodec {
+        case .h264, .vp8:
+            targetSize = CMVideoDimensions(width: 544, height: 480)
+            cropRatio = CGFloat(targetSize.width) / CGFloat(targetSize.height)
+            frameRate = 20
+        case .vp8Simulcast:
+            targetSize = CMVideoDimensions(width: 900, height: 720)
+            cropRatio = CGFloat(targetSize.width) / CGFloat(targetSize.height)
+            frameRate = 24
+        }
+        
+        let captureDevice = CameraSource.captureDevice(position: .front)!
+        let preferredFormat = selectVideoFormatBySize(captureDevice: captureDevice, targetSize: targetSize)
+        preferredFormat.frameRate = min(preferredFormat.frameRate, frameRate)
+        
+        let cropDimensions: CMVideoDimensions
+        
+        if preferredFormat.dimensions.width > preferredFormat.dimensions.height {
+            cropDimensions = CMVideoDimensions(
+                width: Int32(CGFloat(preferredFormat.dimensions.height) * cropRatio),
+                height: preferredFormat.dimensions.height
+            )
+        } else {
+            cropDimensions = CMVideoDimensions(
+                width: preferredFormat.dimensions.width,
+                height: Int32(CGFloat(preferredFormat.dimensions.width) * cropRatio)
+            )
+        }
+        
+        let outputFormat = VideoFormat()
+        outputFormat.dimensions = cropDimensions
+        outputFormat.pixelFormat = preferredFormat.pixelFormat
+        outputFormat.frameRate = 0
+        
+        cameraSource.requestOutputFormat(outputFormat)
+        
+        cameraSource.startCapture(device: captureDevice, format: preferredFormat) { [weak self] _, _, error in
+            guard error == nil else { return }
+            
+            self?.localMediaController.videoCaptureStarted()
+        }
+    }
+        
+    private func selectVideoFormatBySize(captureDevice: AVCaptureDevice, targetSize: CMVideoDimensions) -> VideoFormat {
+        let supportedFormats = Array(CameraSource.supportedFormats(captureDevice: captureDevice)) as! [VideoFormat]
+        
+        for format in supportedFormats {
+            guard
+                format.pixelFormat == .formatYUV420BiPlanarFullRange &&
+                    format.dimensions.width >= targetSize.width &&
+                    format.dimensions.height >= targetSize.height
+                else {
+                    continue
+            }
+            
+            return format
+        }
+        
+        fatalError()
+    }
+}
```

Comment:
@ceaglest I was looking into my next ticket that is supposed to add dynamic capture format. Is this handling it already?